src/main/java/core/basesyntax/service/implementation/FruitShopReportServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.implementation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.service.FruitShopReportService;
+import java.util.Set;
+
+public class FruitShopReportServiceImpl implements FruitShopReportService {
+    private static final String REPORT_HEADER = "fruit, quantity" + System.lineSeparator();
+    private static final String REPORT_SEPARATOR = ", ";
+    private static final String REPORT_NEW_LINE = "\n";
+
+    @Override
+    public String createReport() {
+        String report = REPORT_HEADER;
+        Set<String> fruits = FruitStorage.getKeys();
+        for (String fruit : fruits) {
```

Comment:
you can go through every single map element using EntrySet instead of KeySet and get operation